src/platform/backends/shared/base_virtual_machine_factory.cpp

```diff
@@ -83,27 +84,60 @@ void mp::BaseVirtualMachineFactory::prepare_interface(NetworkInterface& net,
+                                               dest_spec.mem_size,
+                                               dest_spec.disk_space,
+                                               dest_name,
+                                               dest_spec.default_mac_address,
+                                               dest_spec.extra_interfaces,
+                                               dest_spec.ssh_username,
+                                               dest_image,
+                                               cloud_init_path.string().c_str(),
+                                               {},
+                                               {},
+                                               {},
+                                               {}};
+
+    mp::VirtualMachine::UPtr cloned_instance = clone_vm_impl(src_name, src_spec, dest_vm_desc, monitor, key_provider);
+
+    return cloned_instance;
+}
+
+void mp::BaseVirtualMachineFactory::copy_instance_dir_with_essential_files(const fs::path& source_instance_dir_path,
+                                                                           const fs::path& dest_instance_dir_path)
+{
+    assert(fs::exists(source_instance_dir_path) && fs::is_directory(source_instance_dir_path));
+
+    fs::create_directory(dest_instance_dir_path);
+    for (const auto& entry : fs::directory_iterator(source_instance_dir_path))
     {
-        for (const auto& entry : fs::directory_iterator(source_instance_dir_path))
+        // snapshot files are intentionally skipped; .iso file is included for all, .img file here is not relevant
+        // for non-qemu backends.
+        if (entry.path().extension().string() == ".iso" || entry.path().extension().string() == ".img")
         {
-            fs::create_directory(dest_instance_dir_path);
-
-            // 1. Only cloud-init-config.iso file and <image_name>.img file are needed for qemu
-            // 2. Only cloud-init-config.iso file is needed for virutalbox because the rest files are taken care of by
-            // clonevm command
-            // 3. Only cloud-init-config.iso file is needed for hyperv because the rest files are taken
-            // care of by export and import command
-            // By the way, snapshot related files are excluded in all three backends, so as a result we can have an
-            // inclusion file list below which works for all three backends
-            if (entry.path().extension().string() == ".iso" || entry.path().extension().string() == ".img")
-            {
-                const fs::path dest_file_path = dest_instance_dir_path / entry.path().filename();
-                fs::copy(entry.path(), dest_file_path, fs::copy_options::update_existing);
-            }
+            const fs::path dest_file_path = dest_instance_dir_path / entry.path().filename();
+            fs::copy(entry.path(), dest_file_path, fs::copy_options::update_existing);
```

Comment:
Did you mean to use `std::filesystem` (`fs`) here instead of `MP_FILEOPS`?

Reply:
For this particular function, eventually I went with creating actual files in the temporary directory to test. So using MP_FILEOPS does not seem to add value because we no longer need it to mock.